src/test/munit/directory-listener-functional-test-case.xml

```diff
@@ -0,0 +1,396 @@
+            <scheduling-strategy>
+                <fixed-frequency/>
+            </scheduling-strategy>
+        </file:listener>
+        <flow-ref name="storeMessages"/>
+    </flow>
+    <munit:test name="move-to-with-rename-test" description="Verifies that files are moved and renamed after processing">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="source-move-to-with-rename"/>
+        </munit:enable-flow-sources>
+        <munit:behavior>
+            <file:write config-ref="file-list-config" path="matcherless/watchme.txt">
+                <file:content>who watches the watchmen?</file:content>
+            </file:write>
+        </munit:behavior>
+        <munit:execution>
+            <try>
+                <until-successful maxRetries="10" millisBetweenRetries="1000">
+                    <file:read config-ref="file-list-config" path="shared/renamed.txt"/>
+                    <validation:is-not-blank-string/>
+                </until-successful>
+                <set-variable value="#[payload]" variableName="file"/>
+                <try>
+                    <until-successful maxRetries="10" millisBetweenRetries="1000">
+                        <file:read config-ref="file-list-config" path="matcherless/watchme.txt"/>
+                        <validation:is-not-blank-string/>
+                    </until-successful>
+                    <munit-tools:fail/>
+                    <error-handler>
+                        <on-error-continue enableNotifications="false" logException="false" type="MULE:RETRY_EXHAUSTED"/>
+                    </error-handler>
+                </try>
+            </try>
+        </munit:execution>
+    </munit:test>
+
+    <flow name="source-move-to-with-same-file-name">
+        <file:listener config-ref="file-list-config" directory="matcherless" recursive="false" moveToDirectory="shared">
+            <scheduling-strategy>
+                <fixed-frequency/>
+            </scheduling-strategy>
+        </file:listener>
+        <flow-ref name="storeMessages"/>
+    </flow>
+    <munit:test name="move-to-with-same-file-name-test" description="Verifies that files that cannot be moved because a file already exists in the other directory with that name remain untouched">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="source-move-to-with-same-file-name"/>
+        </munit:enable-flow-sources>
+        <munit:behavior>
+            <file:write config-ref="file-list-config" path="shared/watchme.txt">
+                <file:content>who watches the watchmen?</file:content>
+            </file:write>
+            <file:write config-ref="file-list-config" path="matcherless/watchme.txt">
+                <file:content>who watches the watchmen?</file:content>
+            </file:write>
+        </munit:behavior>
+        <munit:execution>
+            <try>
+                <until-successful maxRetries="10" millisBetweenRetries="1000">
```

Comment:
You should first wait some time, and then read the file. It could get moved or deleted if we give it some time, but as we are checking immediately we do not know.